plugins/nf-nomad/src/main/nextflow/nomad/NomadConfig.groovy

```diff
@@ -47,42 +47,65 @@ class NomadConfig {
     }
 
     class NomadClientOpts{
-        final String address
-        final String token
 
-        NomadClientOpts(Map nomadClientOpts){
-            def tmp = (nomadClientOpts.address?.toString() ?: "http://127.0.0.1:4646")
+        private Map<String,String> sysEnv
+        String address
+        String token
+
+        NomadClientOpts(Map nomadClientOpts, Map<String,String> env=null){
+
+            sysEnv = env==null ? new HashMap<String,String>(System.getenv()) : env
+
+            def tmp = (nomadClientOpts.address?.toString() ?: sysEnv.get('NOMAD_ADDR'))
             if( !tmp.endsWith("/"))
                 tmp +="/"
             this.address = tmp + API_VERSION
-            token = nomadClientOpts.token ?: null
+
+            this.token = nomadClientOpts.token ?: sysEnv.get('NOMAD_TOKEN')
+
+            //TODO: Add mTLS properties and env vars
+            // https://developer.hashicorp.com/nomad/docs/commands#mtls-environment-variables
         }
     }
 
     class NomadJobOpts{
-        final boolean deleteOnCompletion
-        final List<String> datacenters
-        final String region
-        final String namespace
-        final String dockerVolume
-        final VolumeSpec volumeSpec
-
-        NomadJobOpts(Map nomadJobOpts){
+
+        //TODO: Reevaluate the job level env vars, which are perhaps better suited to be sourced in clientOpts
+        // and overridden for job defs using process directives.
+        private Map<String,String> sysEnv
+        boolean deleteOnCompletion
+        List<String> datacenters
+        String region
+        String namespace
+        VolumeSpec volumeSpec
+
+        //TODO: Rethink and refactor the usage and design of this variable.
+        // Perhaps better to ask for a specific list of volumes to be mounted
+        boolean mountNextflowHome
```

Comment:
I was thinking to use `volumeSpec` to be mounted as nextflowHome plus a new config `additionalVolumes`  to mount volumes

Reply:
Agreed, I was planning to refactor this anyhow after talking to you this Friday 😉 



`mountNextflowHome` is very hacky 😄